File: old_samples/sample-videochat-webrtc/src/main/java/com/quickblox/sample/groupchatwebrtc/activities/SplashActivity.java
package com.quickblox.sample.groupchatwebrtc.activities;

import android.os.Bundle;

import com.quickblox.sample.core.ui.activity.CoreSplashActivity;
import com.quickblox.sample.core.utils.SharedPrefsHelper;
import com.quickblox.sample.groupchatwebrtc.R;
import com.quickblox.sample.groupchatwebrtc.services.CallService;
import com.quickblox.users.model.QBUser;

public class SplashActivity extends CoreSplashActivity {

    private SharedPrefsHelper sharedPrefsHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        sharedPrefsHelper = SharedPrefsHelper.getInstance();

        if (sharedPrefsHelper.hasQbUser()) {
            startLoginService(sharedPrefsHelper.getQbUser());
            startOpponentsActivity();
            return;
        }

        if (checkConfigsWithSnackebarError()) {
            proceedToTheNextActivityWithDelay();
        }
    }

    @Override
    protected String getAppName() {
        return getString(R.string.splash_app_title);
    }

    @Override
    protected void proceedToTheNextActivity() {
        LoginActivity.start(this);
        finish();
    }

    protected void startLoginService(QBUser qbUser) {
        CallService.start(this, qbUser);
    }

    private void startOpponentsActivity() {
        OpponentsActivity.start(SplashActivity.this, false);
        finish();
    }
}

File: sample-videochat-kotlin/src/main/java/com/quickblox/sample/videochatkotlin/activities/SplashActivity.kt
package com.quickblox.sample.videochatkotlin.activities

import android.content.Intent
import android.os.Bundle
import com.quickblox.sample.core.ui.activity.CoreSplashActivity
import com.quickblox.sample.videochatkotlin.R

/**
 * Created by roman on 4/6/18.
 */
class SplashActivity : CoreSplashActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (checkConfigsWithSnackebarError()) {
            proceedToTheNextActivityWithDelay()
        }
    }

    override fun getAppName(): String {
        return getString(R.string.app_title)
    }

    override fun proceedToTheNextActivity() {
        val intent = Intent(this, MainActivity::class.java)
        this.startActivity(intent)
        finish()
    }
}